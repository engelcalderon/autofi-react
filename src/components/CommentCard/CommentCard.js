import React from 'react';
import { string } from 'prop-types';

import css from './CommentCard.module.css';

const CommentCard = props => {
    const { name, body } = props;

    return (
        <div className={css.root}>
            <h3 className={css.name}>{name}</h3>
            <p className={css.body}>{body}</p>
        </div>
    )
};

CommentCard.defaultProps = {
    name: null,
    body: null
}

CommentCard.propTypes = {
    name: string.isRequired,
    body: string.isRequired
}

export default CommentCard;