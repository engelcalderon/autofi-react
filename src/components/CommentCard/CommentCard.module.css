.root {
    background-color: #f8f8f8;
    padding: 15px;
    margin-bottom: 15px;
    border-radius: 5px;
}

.name {
    margin: 0;
    padding: 5px 0;
}

.body {
    margin: 0;
    padding: 0;
    color: gray;
}