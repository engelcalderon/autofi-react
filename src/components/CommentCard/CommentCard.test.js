import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import CommentCard from './CommentCard';

describe('CommentCard', () => {
    it('matches snapshot', () => {
        const comment = {
            "postId": 1,
            "id": 1,
            "name": "id labore ex et quam laborum",
            "email": "Eliseo@gardner.biz",
            "body": "laudantium enim quasi est quidem magnam voluptate ipsam eos\ntempora quo necessitatibus\ndolor quam autem quasi\nreiciendis et nam sapiente accusantium"
        }
        const tree = toJson(mount(<CommentCard {...comment} />), { mode: 'deep' });
        expect(tree).toMatchSnapshot();
    })
});