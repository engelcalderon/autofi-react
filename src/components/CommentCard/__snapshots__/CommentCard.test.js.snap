// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard matches snapshot 1`] = `
<div
  className="root"
>
  <h3
    className="name"
  >
    id labore ex et quam laborum
  </h3>
  <p
    className="body"
  >
    laudantium enim quasi est quidem magnam voluptate ipsam eos
tempora quo necessitatibus
dolor quam autem quasi
reiciendis et nam sapiente accusantium
  </p>
</div>
`;
